src/ui/ui_settings/routes/__tests__/index_missing.js

```diff
@@ -3,30 +3,19 @@ import expect from 'expect.js';
 import {
   getServices,
   chance,
-  assertDocMissingResponse
+  assertGeneric404Response,
+  waitUntilNextHealthCheck,
 } from './lib';
 
 export function indexMissingSuite() {
-  beforeEach(async function () {
-    const { kbnServer } = getServices();
-    await kbnServer.server.plugins.elasticsearch.waitUntilReady();
-  });
-
-  function getNumberOfShards(index) {
-    return parseInt(Object.values(index)[0].settings.index.number_of_shards, 10);
-  }
-
-  async function getIndex(callCluster, indexName) {
-    return await callCluster('indices.get', {
-      index: indexName,
-    });
-  }
+  beforeEach(waitUntilNextHealthCheck);
 
   async function setup() {
     const { callCluster, kbnServer } = getServices();
     const indexName = kbnServer.config.get('kibana.index');
-    const initialIndex = await getIndex(callCluster, indexName);
 
+    // delete the kibana index and run the test, we have about 2 seconds
+    // before the healthCheck runs again, that SHOULD be enough time
```

Comment:
Oh dear... will this hackery disappear when the health check is gone?

Reply:
yes

Reply:
The whole point of these tests is to verify how the uiSettings behave between healthcheck runs, once it is gone these tests can go too (though we might still want to know how it behaves when there isn't an index template, which will probably require similar hackery)